mindmeld/components/entity_resolver.py

```diff
@@ -569,31 +738,242 @@ def _predict_exact_match(self, entity):
+                cnames_for_syn.append(cname)
+                syn_map[norm_alias] = list(set(cnames_for_syn))
+
+        return {"items": item_map, "synonyms": syn_map}
+
+    def _fit(self, clean):
+        """
+        Fits the resolver model
+
+        Args:
+            clean (bool): If ``True``, deletes existing dump of synonym embeddings file
+        """
+
+        # load model
         try:
-            if self._use_text_rel:
-                scoped_index_name = get_scoped_index_name(
-                    self._app_namespace, self._es_index_name
-                )
-                if not self._es_client.indices.exists(index=scoped_index_name):
-                    self.fit()
-            else:
-                self.fit()
+            _encoding_type = (
+                self.er_config.get("model_settings", {})
+                    .get("encoding_type", "mean")
+            )
+            transformer_model = Transformer(self._sbert_model_pretrained_name_or_abspath)
+            pooling_model = Pooling(transformer_model.get_word_embedding_dimension(),
+                                    pooling_mode_cls_token=_encoding_type == "cls",
+                                    pooling_mode_max_tokens=False,
+                                    pooling_mode_mean_tokens=_encoding_type == "mean",
+                                    pooling_mode_mean_sqrt_len_tokens=False)
+            modules = [transformer_model, pooling_model]
+            self._sbert_model = SentenceTransformer(modules=modules)
+        except OSError:
+            logger.error("Could not initialize the model name through huggingface models; "
+                         "Resorting to model names in sbert.net. "
+                         "Input 'encoding_type', if provided is thus ignored")
+            self._sbert_model = SentenceTransformer(self._sbert_model_pretrained_name_or_abspath)
+
+        # load mappings.json data
+        entity_map = self.resource_loader.get_entity_map(self.type)
+        self._exact_match_mapping = self._process_entity_map(
+            self.type, entity_map
+        )
 
-        except EsConnectionError as e:
-            logger.error(
-                "Unable to connect to Elasticsearch: %s details: %s", e.error, e.info
+        # load embeddings for this data
+        cache_path = self.cache_path(self.pretrained_name)
+        if clean and os.path.exists(cache_path):
+            os.remove(cache_path)
+        if not self._lazy_resolution and os.path.exists(cache_path):
+            self._load()
+            self.dirty = False
+        else:
+            synonyms = [*self._exact_match_mapping["synonyms"]]
+            synonyms_encodings = self._encode(synonyms)
+            self._preloaded_mappings_embs = dict(zip(synonyms, synonyms_encodings))
+            self.dirty = True
+
+        if self.dirty and not self._lazy_resolution:
+            self._dump()
+
+    def _predict(self, entity):
```

Comment:
I think the number of values to be returned should be passed in, with a default being 20